### PR TITLE
Fix missing lint output for organizeDeclarations rule

### PR DESCRIPTION
**Pull request by calda**
This PR fixes an issue where lint violations for the `organizeDeclarations` rule would never be printed when using `--lint`. This could create a confusing experience where the command would output "Source input did not pass lint check" but not tell you while rule the violations came from.

### CODE REVIEW

File: Tests/CommandLineTests.swift [hunk 1]
```diff
@@ -658,4 +658,46 @@ class CommandLineTests: XCTestCase {
...
+
+        XCTAssertEqual(output, [
+            "Running SwiftFormat...",
+            "(lint mode - no files will be changed.)",
+            ":2:1: error: (organizeDeclarations) Organize declarations within class, struct, enum, actor, and extension bodies.",
+            ":3:1: error: (organizeDeclarations) Organize declarations within class, struct, enum, actor, and extension bodies.",
+            ":4:1: error: (organizeDeclarations) Organize declarations within class, struct, enum, actor, and extension bodies.",
+            ":8:1: error: (organizeDeclarations) Organize declarations within class, struct, enum, actor, and extension bodies.",
+            "Source input did not pass lint check.",
+        ])
```
Comment (calda) [inline Tests/CommandLineTests.swift:701]: Before the changes in this PR this test only output:

```
Running SwiftFormat...
(lint mode - no files will be changed.)
Source input did not pass lint check.
```

File: Sources/SwiftFormat.swift [hunk 2]
```diff
@@ -567,18 +567,40 @@ private func applyRules(
...
                     return false
                 }
+
                 return true
             }
+
+            // If all of the changes got filtered out, it's because the only changes are _moves_
+            // that don't modify any individual lines. We still want to emit some list of changes
+            // in this case, so soften the filtering to only exclude duplicates.
+            if !changes.isEmpty, filteredChanges.isEmpty {
```
Comment (calda) [inline Sources/SwiftFormat.swift:591]: The `organizeDeclarations` rule sometimes just moves code without actually modifying the content of the lines. The diffing implementation above was filtering those out. Instead we can use more relaxed filtering implementation when necessary.

**Comment by calda** [comment]
Actually this causes other test failures that I'm not sure how to fix, so I'm just going to close this for now.

**Comment by nicklockwood** [comment]
Perhaps what's needed is a more sophisticated model for change tracking, where we distinguish between edits and moves? Then if the change type was "edit" but the line hasn't changed then it wouldn't be logged, but if it's "move" then it still would be?

**Comment by calda** [comment]
Thanks for the suggestion @nicklockwood, that sort of approach worked much better!

I added a new `formatter.moveTokens` helper that moves tokens and tags the `Formatter.Change` as a move that shouldn't be filtered out later. Since `organizeDeclarations` works by replacing `formatter.tokens` all at once, I also added a `formatter.updateAllTokens` helper that diffs the lines and tags the `Formatter.Change` for any moved lines as a move.

I also added a check for this issue in the `testFormatting` helper, which highlighted that the `docCommentsBeforeAttributes` rule also had this issue. Changing that rule to use `formatter.moveTokens` fixed it.

**Comment by nicklockwood** [comment]
Amazing! Thanks - lgtm

**Pull request merged**